curriculum/challenges/english/25-front-end-development/lab-factorial-calculator/66c07238b01053abaf812065.md

```diff
@@ -0,0 +1,91 @@
+---
+id: 66c07238b01053abaf812065
+title: Build a Factorial Calculator
+challengeType: 14
+dashedName: build-a-factorial-calculator
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+A factorial is the product of an integer and all the integers below it. For example, the factorial of `5` is `5 * 4 * 3 * 2 * 1 = 120`. In this lab, you will create a factorial calculator that takes a number from the user and calculates the factorial of that number.
+
+1. You should declare a variable `num` and assign it the value of `5`.
+   
+2. Declare a variable `result` and assign it the value of `1`.
+   
+3. Using a loop, loop through all numbers from `1` to `num` (inclusive) and for each number, multiply the `result` variable by the current number and assign the result to the `result` variable. You can choose to use  either a `for` loop, `while` loop or `do...while` loop here.
+   
+4. You should store the factorial of `num` in the variable `resultMsg` in the format `Factorial of [num] is [result]`.
+
+5. You should output the value of `resultMsg` to the console.
+   
+# --hints--
+
+You should have a `num` variable.
+
+```js
+assert.isDefined(num);
+```
+
+You should assign `5` to the `num` variable
+
+```js
+assert.equal(num, 5);
+```
+
+You should have a `result` variable.
+
+```js
+assert.isDefined(result);
+```
+
+You should assign `1` to the `result` variable.
+
+```js
+assert.match(__helpers.removeJSComments(code), /result\s*=\s*1\s*;?\s?$/m);
```

Comment:
do we need a regex here?
assert.equal should work since result is a global variable right?

Reply:
`assert.equal` didn't work here as expected, maybe because it is reassigned later on.